### ISSUE TITLE
resolve --transitive-dependencies should preserve dependency order

### ISSUE THREAD

**Issue by retnuh** [dhall-lang/dhall-lang#823]
It would be nice it running `dhall resolve --transitive-dependencies` preserved the dependency order, so that when a file/url is printed out, we know that all it's dependencies have already been printed out.

This make it more useful for scripts that perform a freeze as part of a build/release process; we can just use `resolve` to get a list of files, in the correct freezing order.

**Comment by sjakobi** [comment]
Wouldn't it be even nicer if `dhall freeze` could freeze an entire project out of the box? With a `--transitive` flag or something like that?

**Comment by Gabriel439** [comment]
@sjakobi: I think we should do both.  Nix does something similar and having dependencies in post-order really helps when debugging dependency trees.

I have a change for just ordering the dependencies as requested up here: https://github.com/dhall-lang/dhall-haskell/pull/1539

**Comment by Gabriel439** [comment]
Actually, adding `dhall freeze` support for transitively following imports could present UX challenges

For example:

* attempting to freeze transitive dependencies that you don't have permission to modify
* freezing dependencies outside of your current project by mistake

**Comment by retnuh** [comment]
maybe an option for following local file based dependencies, but not network dependencies?

**Comment by Gabriel439** [comment]
@retnuh: Those issues still apply to only file-based dependencies

**Comment by retnuh** [comment]
Yes it does start to become tricky UX wise - maybe specify a list of allowed root paths?  I dunno, I'm no UX expert.  But now we've got `resolve` to help with our own tools at least 😉 🎉

**Comment by sjakobi** [comment]
I'll move this to the dhall-haskell repo, where the relevant code lives.
